piptools/writer.py

```diff
@@ -89,14 +90,15 @@ def _iter_lines(self, results, reverse_dependencies, primary_packages):
     def write(self, results, reverse_dependencies, primary_packages):
         with ExitStack() as stack:
             f = None
-            if not self.dry_run:
+            if not self.dry_run and self.dst_file != '-':
                 f = stack.enter_context(AtomicSaver(self.dst_file))
 
             for line in self._iter_lines(results, reverse_dependencies, primary_packages):
-                log.info(line)
```

Comment:
@cjerdonek 
> shouldn't the log be going to stderr like most logging does?

Agreed.  Do you want to create an issue for it?
But then this log line is still a bit too verbose (move it to debug? / add a prefix like "writing line"?).